x-pack/plugins/monitoring/public/application/setup_mode/setup_mode_renderer.js

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Fragment } from 'react';
```

Comment:
This seems like it's forked from https://github.com/elastic/kibana/blob/master/x-pack/plugins/monitoring/public/components/renderers/setup_mode.js

Do you think we could use the legacy shims to avoid the fork? Or if not should we maybe also fork https://github.com/elastic/kibana/blob/master/x-pack/plugins/monitoring/public/components/renderers/setup_mode.test.js ?

Reply:
Yes, it's forked from that file. I couldn't use the legacy shims because there is no scope mock in the Legacy class and because the global state context doesn't exist in the Angular work and it's needed here.

But good point with forking the tests, I'll take a look now to see if it's easy to adapt them to the changes made 🤞

Reply:
I didn't fork tests yet. There are some snapshot tests that I think it would be interesting to run against (I could also copy that, but I think is too much copy).